ecosystem-modules/starport/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+		NoticeAlreadyInvoked,
+		/// Only specific Admin account is able to send Notice to be invoked
+		InvalidNoticeInvoker,
+		/// The Admin account does not have enough asset for the Unlock operation.
+		InsufficientAssetToUnlock,
+		/// Not enough authorities have signed this notice for it to be effective.
+		InsufficientValidNoticeSignatures,
+		/// Too many Authorities.
+		ExceededMaxNumberOfAuthorities,
+		/// Authorities cannot be empty
+		AuthoritiesListCannotBeEmpty,
+		/// Only the current Gateway Admin can change the current Gateway Admin.
+		InvalidGatewayAdminUpdateCaller,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	#[pallet::metadata(Balance = "Balance", T::AccountId = "AccountId")]
+	pub enum Event<T: Config> {
+		/// User has locked some asset and uploaded them into Compound.
+		AssetLockedTo(CurrencyId, Balance, T::AccountId),
+
+		/// The user has unlocked some asset and downloaded them back into Acala.
+		AssetUnlocked(CurrencyId, Balance, T::AccountId),
+
+		/// The list of authorities has been updated.
+		GatewayAuthoritiesChanged,
+
+		/// The supply cap for an asset has been updated.
+		SupplyCapSet(CurrencyId, Balance),
+
+		/// The future yield for CASH is set.
+		FutureYieldSet(Balance, CashYieldIndex, Moment),
+
+		/// Gateway's Admin Account has been updated.
+		GatewayAdminUpdated(T::AccountId),
+	}
+
+	#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq)]
+	pub struct GatewayNotice<AccountId> {
+		pub id: u64,
+		pub payload: GatewayNoticePayload<AccountId>,
+	}
+
+	impl<AccountId> GatewayNotice<AccountId> {
+		pub fn new(id: u64, payload: GatewayNoticePayload<AccountId>) -> Self {
+			Self { id, payload }
+		}
+	}
+
+	#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq)]
+	pub enum GatewayNoticePayload<AccountId> {
+		/// Update the current supply cap for an asset. Only assets that have spare supplies.
+		/// can be locked or uploaded to the Compound chain.
+		SetSupplyCap(CurrencyId, Balance),
+
+		/// Update the current set of authorities who sign Notices.
+		ChangeAuthorities(Vec<CompoundAuthoritySignature>),
+
+		/// Unlock or download assets from Compound chain back into Acala chain.
+		Unlock(CurrencyId, Balance, AccountId),
+
+		/// Set the future yield for the Cash asset.
+		/// Parameters: uint128 nextCashYield, uint128 nextCashYieldIndex, uint nextCashYieldStart
+		SetFutureYield(Balance, CashYieldIndex, Moment),
+	}
+
+	/// Stores the amount of supplies that are still available to be uploaded for each asset type.
+	#[pallet::storage]
+	#[pallet::getter(fn supply_caps)]
+	pub type SupplyCaps<T: Config> = StorageMap<_, Twox64Concat, CurrencyId, Balance, ValueQuery>;
+
+	/// Stores the Hash of Notices that have already been invoked. PrevenTwox64Concatts
+	/// double-invocation.
+	#[pallet::storage]
+	#[pallet::getter(fn invoked_notice_hashes)]
+	pub type InvokedNoticeHashes<T: Config> = StorageMap<_, Identity, H256, (), ValueQuery>;
```

Comment:
```suggestion
	pub type InvokedNoticeHashes<T: Config> = StorageMap<_, Identity, H256, (), OptionQuery>;
```

Reply:
also check if OptionQuery should be used for other storages. we should default o use OptionQuery except for the case missing value can be same as default value